src/extra/components/xgm.py

```diff
@@ -0,0 +1,353 @@
+from enum import Enum
+
+import numpy as np
+
+from extra_data import SourceData, KeyData
+
+
+class PropertyGroup(Enum):
+    """Enum representing the different 'groups' of XGM properties.
+
+    XGMs in SASE 1 and SASE 3 hold data for both SASEs, as well as their
+    combined data in the default properties (e.g. `data.intensityTD`). To
+    simplify access to these, this enum defines groups for 'SASE-resolved'
+    properties. However, this is purely an internal abstraction. To users we
+    only expose `sase` arguments in the XGM methods (which may be `0` to refer
+    to the main group).
+    """
+    MAIN = 0
+    SASE1 = 1
+    SASE3 = 3
+
+class XGM:
+    """Interface to an XGM (X-ray Gas Monitor).
+
+    Example usage in a Jupyter notebook:
+    ```python
+            -----------------------------------------------------------
+    In [1]: |xgm = XGM(run) # This will work with a single XGM        |
+            |xgm                                                      |
+            -----------------------------------------------------------
+    Out[1]: <XGM for SA1_XTD2_XGM/XGM/DOOCS at 9.3 KeV>
+
+            ----------------------------------------------------------
+    In [2]: |# With multiple XGMs, one needs to be chosen explicitly |
+            |xgm = XGM(run, run.alias["spb-xgm"])                    |
+            ----------------------------------------------------------
+    Out[2]: <XGM for SPB_XTD9_XGM/XGM/DOOCS at 9.3 KeV>
+    ```
+    """
+
+    # There are three different XGM device classes, though all are very
+    # similar. DoocsXGM is the main one, and DoocsXGMD and DoocsXGMReduced are
+    # used by FXE and SQS respectively. For more information see the original MR
+    # to create them: https://git.xfel.eu/karaboDevices/doocsDevices/-/merge_requests/117
+    DEVICE_CLASSES = ["DoocsXGM", "DoocsXGMD", "DoocsXGMReduced"]
+
+    def __init__(self, run, device=None, default_sase=None):
```

Comment:
At HED it is typical to always record 2 XGM, one right after the ondulators and one just before the experimental hutch. (I would assume all instrument do something similar).

I think it may be useful to have an easy filtering in the device name, so that one could do: `XGM(run, "SA2")`/`XGM(run, "HED")` or `XGM(run, "XTD1")`/`XGM(run, "XTD6")` without having to type the complete device name or rely on an alias.

Reply:
Added in a8e7c50, now it accepts case-insensitive unique substrings. I also added support for recognizing alias names for convenience.